src/engine/loopingcontrol.cpp

```diff
@@ -1153,22 +1114,20 @@ void LoopingControl::slotBeatJumpBackward(double pressed) {
 }
 
 void LoopingControl::slotLoopMove(double beats) {
-    if (!m_pTrack || !m_pBeats) {
+    if (m_pTrack == nullptr || m_pBeats == nullptr || beats == 0) {
         return;
     }
     LoopSamples loopSamples = m_loopSamples.getValue();
     if (loopSamples.start == kNoTrigger || loopSamples.end == kNoTrigger) {
         return;
     }
 
-    double dPosition = getCurrentSample();
-    double dBeatLength;
-    if (BpmControl::getBeatContext(m_pBeats, dPosition,
-                                   NULL, NULL, &dBeatLength, NULL)) {
+    if (BpmControl::getBeatContext(m_pBeats, getCurrentSample(),
+                                   nullptr, nullptr, nullptr, nullptr)) {
         int old_loop_in = loopSamples.start;
         int old_loop_out = loopSamples.end;
-        int new_loop_in = old_loop_in + (beats * dBeatLength);
-        int new_loop_out = old_loop_out + (beats * dBeatLength);
+        int new_loop_in = m_pBeats->findNBeatsFromSample(old_loop_in, beats);
+        int new_loop_out = m_pBeats->findNBeatsFromSample(old_loop_out, beats);
```

Comment:
This can be a false positive, because of the fractional beat position. 

Lets say a beat is 50.5 beats, rounded to the nearest sample you have one 50 frame loop and one 51 frame loop. (delta 2 samples)

This means also, we are 10 frames ahead/early after 20 loop cycles and the reason why a floating point transport engine makes sense. 

It seams to be out of scope to add more floating point looping to this PR, so I would just allow an offset of two samples in the test.